tdp_vars_defaults/hadoop/hadoop.yml

```
@@ -178,10 +178,11 @@ jmx_exporter:
   lowercaseOutputLabelNames: false
 
 # Ranger KMS url 
-ranger_kms_url: "kms://{{ ranger_kms_hosts }}:9393/kms"
-ranger_kms_hosts: |-
+ranger_kms_url: "{% if 'ranger_kms' in groups and groups['ranger_kms'] %}kms://{{ ranger_kms_hosts| trim }}:9393/kms{% else %}NONE{% endif %}"
+ranger_kms_hosts: "{% if 'ranger_kms' in groups and groups['ranger_kms'] %}
     {{ groups['ranger_kms'] | 
       map('tosit.tdp.access_fqdn', hostvars) |
       map('regex_replace', '^', 'https@') |
       list |
       join(';') }}
+    {% else %}NONE{% endif %}"
```

```suggestion
ranger_kms_hosts: |-
  {% if 'ranger_kms' in groups and groups['ranger_kms'] %}
  {{ groups['ranger_kms'] | 
    map('tosit.tdp.access_fqdn', hostvars) |
    map('regex_replace', '^', 'https@') |
    join(';') }}{% else %}NONE{% endif %}
```

You can set the variable like this if you don't want extra new lines